addons/stock_account/models/stock.py

```diff
@@ -377,7 +377,12 @@ def product_price_update_before_done(self, forced_qty=None):
         tmpl_dict = defaultdict(lambda: 0.0)
         # adapt standard price on incomming moves if the product cost_method is 'average'
         std_price_update = {}
-        for move in self.filtered(lambda move: move.location_id.usage in ('supplier', 'production') and move.product_id.cost_method == 'average'):
+        for move in self.filtered(
+                lambda move: move.location_id.usage in ('supplier', 'production')
```

Comment:
maybe you kept the original condition on purpose, but you could use move._is_in here. the side effect is that "receiving" products from an inventory adjustment will now update the avco, but imo this would be the right behaviour (updating the avco at each IN)